shared/profile/user/index.tsx

```diff
@@ -403,11 +403,13 @@ class User extends React.Component<Props, State> {
 
   _renderWebOfTrust = ({item}) =>
     item.type === 'IKnowThem' ? (
-      <Kb.Box2 key="iknowthem" direction="horizontal" fullWidth={true} style={styles.knowThemBox}>
-        <Kb.Button key="iknowthembtn" onClick={this.props.onIKnowThem} type="Default" label={item.text}>
-          <Kb.Icon type="iconfont-proof-good" style={styles.knowThemIcon} />
-        </Kb.Button>
-      </Kb.Box2>
+      this.props.onIKnowThem && (
+        <Kb.Box2 key="iknowthem" direction="horizontal" fullWidth={true} style={styles.knowThemBox}>
+          <Kb.Button key="iknowthembtn" onClick={this.props.onIKnowThem} type="Default" label={item.text}>
+            <Kb.Icon type="iconfont-proof-good" style={styles.knowThemIcon} />
+          </Kb.Button>
+        </Kb.Box2>
+      )
```

Comment:
Generally speaking, it would be better to use a boolean or string to conditionally render a react component than a function (even though they're both truthy).

If you can plumb `promptForVouce` through, I would suggest using that here

Reply:
this component has a couple of other places where we're looking at the presence of a dispatch function to decide whether or not to show the thing that it does. i agree that a boolean would be nicer for that, but it means plumbing through 3 or 4 additional booleans, which seems sub-optimal to me.

Reply:
I saw that after making this comment. It's fine to leave it for now then.

Given the recency of writing the web of trust components and how small they are now, moving the code from 'container.tsx` to `index.tsx` as hookds might help out by removing the need to render based on dispatchProps callbacks.

Reply:
yeah - i think this is not un-compelling. i'm in the middle of another related ticket (adding the delete button). i'll do a bunch of cleanup in that ticket, including moving it to one index file with hooks. that cool?

Reply:
Sounds dandy! 🌻

Reply:
it's definitely nicer in one file. you're for sure right.